js/ui/iconGridLayout.js

```diff
@@ -94,30 +100,62 @@ const IconGridLayout = new Lang.Class({
         this._iconTree = iconTree;
     },
 
-    _getDefaultIcons: function() {
-        let iconTree = null;
+    _loadConfigJsonString: function(dir, base) {
+        let jsonString = null;
         let defaultFiles = GLib.get_language_names()
             .filter(function(name) {
                 return name.indexOf('.') == -1;
             })
             .map(function(name) {
-                let path = GLib.build_filenamev([DEFAULT_CONFIGS_DIR,
-                                                 DEFAULT_CONFIG_NAME_BASE + '-' + name + '.json']);
+                let path = GLib.build_filenamev([dir, base + '-' + name + '.json']);
                 return Gio.File.new_for_path(path);
             })
             .some(function(defaultsFile) {
                 try {
                     let [success, data] = defaultsFile.load_contents(null, null,
                                                                      null);
-                    iconTree = Json.gvariant_deserialize_data(data.toString(), -1,
-                                                              'a{sas}');
+                    jsonString = data.toString();
                     return true;
                 } catch (e) {
                     // Ignore errors, as we always have a fallback
                 }
-
                 return false;
             });
+        return jsonString;
+    },
+
+    _mergeJsonStrings: function(base, prepend, append) {
+        let baseNode = JSON.parse(base)
+        let prependNode = null;
+        let appendNode = null;
+        if (prepend) {
+            prependNode = JSON.parse(prepend);
+	}
+        if (append) {
+            appendNode = JSON.parse(append);
+	}
+        for (var key in baseNode) {
+            if (prependNode && prependNode[key]) {
+                baseNode[key] = prependNode[key].concat(baseNode[key]);
+            }
+            if (appendNode && appendNode[key]) {
+                baseNode[key] = baseNode[key].concat(appendNode[key]);
+	    }
+        }
+        return JSON.stringify(baseNode);
+    },
+
+    _getDefaultIcons: function() {
+        let iconTree = null;
+        let mergedJson = null;
```

Comment:
No need to declare these variables in a separate line/initialize them to `null` beforehand